test/services/bill-runs/two-part-tariff/allocate-returns-to-charge-element.service.test.js

```diff
@@ -144,6 +144,53 @@ describe('Allocate Returns to Charge Element Service', () => {
         })
       })
 
+      // NOTE It was found during testing that if the individual return line volumes are greater than the charge
+      // reference authorised volume, which in turn is greater than the element authorised volume. That the element was
+      // being over allocated. In this scenario 3.5 would have been allocated to the element. This scenario has
+      // therefore been created to test that this does not happen.
+      describe('with a reference authorised to 3.5, element to 3, matched to a return with line volumes of 4', () => {
+        beforeEach(() => {
+          testData = _generateTestData()
+          testData.chargeReference.volume = 3.5
+          testData.chargeElement.authorisedAnnualQuantity = 3
+        })
+
+        it('allocates 3 to the charge reference', () => {
+          const { chargeElement, chargeReference, matchingReturns } = testData
+
+          AllocateReturnsToChargeElementService.go(chargeElement, matchingReturns, chargePeriod, chargeReference)
+
+          expect(chargeReference.volume).to.equal(3.5)
+          expect(chargeReference.allocatedQuantity).to.equal(3)
+        })
+
+        it('allocates 3 to the charge element', () => {
+          const { chargeElement, chargeReference, matchingReturns } = testData
+
+          AllocateReturnsToChargeElementService.go(chargeElement, matchingReturns, chargePeriod, chargeReference)
+
+          expect(chargeElement.authorisedAnnualQuantity).to.equal(3)
+          expect(chargeElement.allocatedQuantity).to.equal(3)
+          expect(chargeElement.returnLogs[0].allocatedQuantity).to.equal(3)
+        })
+
+        it('allocates 10 from the return log', () => {
```

Comment:
Think this is meant to be 'allocates 3 from the return log' not 10? :)

Reply:
You are correct, I'll changer that